internal/app/handlers/publicationviewing/show.go

```diff
@@ -0,0 +1,162 @@
+package publicationviewing
+
+import (
+	"net/http"
+
+	"github.com/ugent-library/biblio-backend/internal/app/displays"
+	"github.com/ugent-library/biblio-backend/internal/bind"
+	"github.com/ugent-library/biblio-backend/internal/models"
+	"github.com/ugent-library/biblio-backend/internal/render"
+	"github.com/ugent-library/biblio-backend/internal/render/display"
+	"github.com/ugent-library/biblio-backend/internal/validation"
+)
+
+type YieldShow struct {
+	Context
+	PageTitle    string
+	ActiveNav    string
+	ActiveSubNav string
+	SearchArgs   *models.SearchArgs
+}
+
+type YieldShowDescription struct {
+	Context
+	ActiveSubNav          string
+	SearchArgs            *models.SearchArgs
+	DisplayDetails        *display.Display
+	DisplayConference     *display.Display
+	DisplayAdditionalInfo *display.Display
+}
+
+type YieldShowContributors struct {
+	Context
+	ActiveSubNav string
+	SearchArgs   *models.SearchArgs
+}
+
+type YieldShowDatasets struct {
+	Context
+	ActiveSubNav    string
+	SearchArgs      *models.SearchArgs
+	RelatedDatasets []*models.Dataset
+}
+
+var allowedSubNavs = []string{
+	"description",
+	"files",
+	"contributors",
+	"datasets",
+}
+
+func (h *Handler) Show(w http.ResponseWriter, r *http.Request, ctx Context) {
+	searchArgs := models.NewSearchArgs()
+	if err := bind.Request(r, searchArgs); err != nil {
+		render.BadRequest(w, r, err)
+		return
+	}
+
+	// TODO bind and validate
+	activeSubNav := r.URL.Query().Get("show")
+	if !validation.InArray(allowedSubNavs, activeSubNav) {
+		activeSubNav = "description"
+	}
+
+	render.Wrap(w, "layouts/default", "publication/show_page", YieldShow{
+		Context:      ctx,
+		PageTitle:    ctx.T("publication.page.show.title"),
+		ActiveNav:    "publications",
+		ActiveSubNav: activeSubNav,
+		SearchArgs:   searchArgs,
+	})
+}
+
+func (h *Handler) ShowDescription(w http.ResponseWriter, r *http.Request, ctx Context) {
+	searchArgs := models.NewSearchArgs()
+	if err := bind.Request(r, searchArgs); err != nil {
+		render.BadRequest(w, r, err)
+		return
+	}
+
+	render.Render(w, "publication/show_description", YieldShowDescription{
+		Context:               ctx,
+		ActiveSubNav:          "description",
+		SearchArgs:            searchArgs,
+		DisplayDetails:        displayDetails(ctx),
```

Comment:
i'll move the switch to the displays package

Reply:
switch?

Reply:
displayDetails()